### ISSUE TITLE
[BUG] Hugging Face based estimators are failing in CI

### ISSUE THREAD

**Issue by yarnabrina** [sktime/sktime#7492]
Originally observed in CI of #7488 (Ref. https://github.com/sktime/sktime/actions/runs/12204522792/job/34049834324?pr=7488), there were a lot of failures during import of `transformers`.

> RuntimeError: Failed to import transformers.trainer because of the following error (look up to see its traceback):
cannot import name 'EncoderDecoderCache' from 'transformers' (/opt/hostedtoolcache/Python/3.9.20/x64/lib/python3.9/site-packages/transformers/init.py)

Upon debugging the logs, it seeme to be caused by unprotected imports at top module level, and @fkiraly suggested over Discord to move them inside methods. #7491 attempts that fix in two modules, but then it is observed thet it is generated from `peft` imports as part of tests parameters of `HFTransformersForecaster` as well, who are importing `transformers` internally.

FYI @benHeid @geetu040 @fkiraly

**Comment by fkiraly** [comment]
The notebook run is also failing, in the global forecasting notebook.

I am also noticing there is no upper bound on `huggingface-hub`, but there is on `transformers`.

What explains the error? A change in which package precisely?

**Comment by yarnabrina** [comment]
> What explains the error? A change in which package precisely?

I have no idea at all. I tried to install `transformers` locally and it is working fine. It did resolve to latest `transformers` by default, so I then manually installed `4.40.2` as CI is doing and that also passes. my guess is it is side effect from some other package, or something is wrong in CI cache somewhere (whought why or how no idea).

**Comment by fkiraly** [comment]
If it is two test cases, I would suggest to isolate imports around them. As long as the import issue is fixed, we can skip the individual tests.

**Comment by fkiraly** [comment]
I think the culprit is recently released `peft 0.14.0`, which changed imports without warning.

Temporary fix attempted here: https://github.com/sktime/sktime/pull/7495

**Comment by fkiraly** [comment]
> @fkiraly if a PR modifies a package in pyproject and no other changes, does CI run tests of estimators which do not have that package in python_dependencies tag?

No, it does not - currently. That is why we should populate the dependency tags properly.
